### ISSUE TITLE
False positive UnusedPrivateClass in RC11

### ISSUE THREAD

**Issue by scottkennedy** [arturbosch/detekt#1347]
```kt
package com.example

class Test {
    val items = Item.values().map { it.text }.toList()
}

private enum class Item(val text: String) {
    A("A"),
    B("B"),
    C("C")
}
```

## Expected Behavior
No issues reported, because `Item` is clearly used by `Test`.

## Observed Behavior
"Private class Item is unused."

## Steps to Reproduce
See above code.

## Your Environment
* Version of detekt used: RC11
* Version of Gradle used (if applicable): 4.10.2
* Operating System and version: Various
